src/routes/_surfaces/workspace/explore/time-controls/TimeRangeNameSelector.svelte

```diff
@@ -0,0 +1,117 @@
+<script lang="ts">
+  import type {
+    TimeRangeName,
+    TimeSeriesTimeRange,
+  } from "$common/database-service/DatabaseTimeSeriesActions";
+  import CaretDownIcon from "$lib/components/icons/CaretDownIcon.svelte";
+  import Menu from "$lib/components/menu/Menu.svelte";
+  import MenuItem from "$lib/components/menu/MenuItem.svelte";
+  import FloatingElement from "$lib/components/tooltip/FloatingElement.svelte";
+  import { getMetricsExplorerById } from "$lib/redux-store/explore/explore-readables";
+  import { onClickOutside } from "$lib/util/on-click-outside";
+  import { tick } from "svelte";
+  import {
+    getDefaultTimeRangeName,
+    getSelectableTimeRangeNames,
+    makeTimeRanges,
+    prettyFormatTimeRange,
+  } from "./time-range-utils";
+
+  export let metricsDefId: string;
+  export let selectedTimeRangeName: TimeRangeName;
+  export let onSelectTimeRangeName: (timeRangeName: TimeRangeName) => void;
+
+  $: metricsExplorer = getMetricsExplorerById(metricsDefId);
+
+  let selectableTimeRanges: TimeSeriesTimeRange[];
+
+  const getSelectableTimeRanges = (
+    allTimeRangeInDataset: TimeSeriesTimeRange
+  ) => {
+    // TODO: replace this with a call to the `/meta` endpoint, once available.
+    const selectableTimeRangeNames = getSelectableTimeRangeNames(
+      allTimeRangeInDataset
+    );
+    const selectableTimeRanges = makeTimeRanges(
+      selectableTimeRangeNames,
+      allTimeRangeInDataset
+    );
+    return selectableTimeRanges;
+  };
+  $: if ($metricsExplorer?.allTimeRange) {
+    selectableTimeRanges = getSelectableTimeRanges(
```

Comment:
Any idea why this was moved back to a reactive statement? Makes sense to calculate this when `allTimeRange` is updated.

Reply:
A couple reasons:
- I think semantically it makes sense for the component to be in charge of both data fetching & data rendering. That keeps all the logic encapsulated in one place. This call to `getSelectableTimeRanges` will be replaced by a call to `/meta`, and when we do that, I think that call should live in this component.
- I had to refactor these Time Controls components and there were some merge conflicts, so I had to pick an approach and it was faster for me to just stick with the code on my branch.

Reply:
Ok this reverted my reactivity changes in dashboard. We 1st see call to full time range then for the default time range.